stan/math/rev/fun/log_sum_exp.hpp

```diff
@@ -22,8 +22,8 @@ class log_sum_exp_vv_vari : public op_vv_vari {
   log_sum_exp_vv_vari(vari* avi, vari* bvi)
       : op_vv_vari(log_sum_exp(avi->val_, bvi->val_), avi, bvi) {}
   void chain() {
-    avi_->adj_ += adj_ * calculate_chain(avi_->val_, val_);
```

Comment:
Would it be possible to get rid of calculate_chain everywhere? It seems like a strange function.

Also used here (which should be fixed as part of this pull):
https://github.com/stan-dev/math/blob/b5133581d24162e75ca5d04a5a39d43205707767/stan/math/rev/fun/log_sum_exp.hpp#L37

In stan/math/rev/fun/log_diff_exp.hpp and in stan/math/rev/fun/log1p_exp.hpp.